src/helpers/routes.js

```diff
@@ -14,71 +14,59 @@ import ConnectedRequestPriceComponent from '../connectedComponents/pipeline/Conn
 import ConnectedRequestReviewComponent from '../connectedComponents/pipeline/ConnectedRequestReview';
 import ConnectedRequestCompleteComponent from '../connectedComponents/pipeline/ConnectedRequestComplete';
 import HomePageComponent from '../components/pages/Homepage/HomePage';
-import { App } from '../components/App';
+import App from '../components/App';
 import reducers from '../reducers/reducers';
 import Styles from '../styles/App';
 
-export class Routes extends Component {
-  constructor() {
-    super();
-
-    const store = createStore(reducers, applyMiddleware(thunk));
-
-    sessionService.initSessionService(store);
-
-    this.state = {
-      store,
-    };
-
-    store.subscribe(() => {
-      console.log('store.getState()', store.getState());
-    });
-  }
-  render() {
-    return (
-      <Provider store={this.state.store}>
-        <BrowserRouter className="router">
-          <Route
-            render={props => (
-              <div style={Styles.div} {...props}>
-                <ConnectedNavBarComponent style={Styles.navBar} {...props} />
-                <Switch>
-                  <Route exact path="/test" component={App} />
-                  <Route exact path="/pipeline" component={ConnectedRequestSelection} />
-                  <Route exact path="/HomePage" component={HomePageComponent} />
-                  <Route
-                    exact
-                    path="/pipeline/requestInformation"
-                    component={ConnectedRequestInformationComponent}
-                  />
-                  <Route
-                    exact
-                    path="/pipeline/requestLocation"
-                    component={ConnectedRequestLocationComponent}
-                  />
-                  <Route
-                    exact
-                    path="/pipeline/requestPrice"
-                    component={ConnectedRequestPriceComponent}
-                  />
-                  <Route
-                    exact
-                    path="/pipeline/requestReview"
-                    component={ConnectedRequestReviewComponent}
-                  />
-                  <Route
-                    exact
-                    path="/pipeline/requestComplete"
-                    component={ConnectedRequestCompleteComponent}
-                  />
-                  <Route path="/*" component={Error404} />
-                </Switch>
-              </div>
-            )}
-          />
-        </BrowserRouter>
-      </Provider>
-    );
-  }
+function Routes() {
+  const store = createStore(reducers, applyMiddleware(thunk));
```

Comment:
Does this work without storing `store` inside of the component state?

Reply:
I mean the store is still being updated as it was before so I don't think theres anything wrong with just passing in store as a function .

Reply:
The question I had was about where the store lives after this function runs. Does a new one get generated on every render? Or does something else keep `store` from being recreated? The reason I asked is because every time I have worked with Redux, I have had to make `store` a state variable on the component to keep it from being recreated.

Reply:
I don't think that store state is being used though, since we aren't updating it in any way.

Reply:
So for some background, anytime a prop updates in a component, a rerender occurs. Since this is a functional component, the function will run again every time the component gets an update. Since this component is responsible for routing, it will receive an update **at least** every time the user goes to a new page. This means that, for example, a user would login, be redirected to a new page, and would be instantly logged out again. This is what I fear will happen if we don't maintain the original `store`, we will get a new instance, instantiated with whatever default state we wanted in the reducers. We can avoid this by storing the `store` inside of the component state.

Reply:
ok i can do that

Reply:
done

Reply:
Would it work if we checked the existing state via the hooks and used that to update the new state?

Reply:
@btrettin you still need to call `useState` on `store`. Just don't assign what returns from the `useState` call to a variable. We don't need that one. We just need it inside of the component state.

Reply:
Do you mean use:

<Provider store={useState(store)}>

I think it'd be good if you just type out what you mean

Reply:
``` javascript
useState(store);
```

Just don't assign the result to a local variable. We only need `store` inside of the component state. We don't need to reference it again later.

Reply:
1. Are we using the sessionService?

2. Do we need state for the component or can we just call store={createStore(reducers, applyMiddleware(thunk))} in the provider, because that seems to workwith mainintaing the store on redirects. otherwhise we can also do

const[store]= createStore(reducers, applyMiddleware(thunk))

and then call store={store} in the provider.

Reply:
Lmk what you think of the changes made.

Reply:
After doing some searching, it looks like `store` doesn't need to be kept in a component state for it to be maintained. This looks good then 👍